inst/include/child_model_simulation.hpp

```diff
@@ -246,6 +235,444 @@ void add_child_grad(int time_step,
+void run_child_art_mortality(int time_step,
+                             const Parameters<ModelVariant, real_type> &pars,
+                             const State<ModelVariant, real_type> &state_curr,
+                             State<ModelVariant, real_type> &state_next,
+                             IntermediateData<ModelVariant, real_type> &intermediate) {
+  static_assert(ModelVariant::run_child_model,
+                "run_hiv_child_infections can only be called for model variants where run_child_model is true");
+  constexpr auto ss = StateSpace<ModelVariant>().base;
+  constexpr auto hc_ss = StateSpace<ModelVariant>().children;
+  const auto cpars = pars.children.children;
+
+
+  for(int hd = 0; hd < hc_ss.hc1DS; ++hd) {
+    for(int a = 0; a < pars.base.options.p_idx_fertility_first; ++a) {
+      for(int s = 0; s <ss.NS; ++s) {
+        intermediate.children.hc_death_rate = 0.0;
+        intermediate.children.hc_art_grad(0, hd, a, s) = 0.0;
+        if(a < hc_ss.hc2_agestart){
+          if(state_next.children.hc1_art_pop(0, hd, a, s) > 0){
+            intermediate.children.hc_death_rate = cpars.hc_art_mort_rr(0, a, time_step) * 0.5 * (cpars.hc1_art_mort(hd, 0, a) + cpars.hc1_art_mort(hd, 1, a));
+            state_next.children.hc1_art_aids_deaths(0, hd, a, s) = intermediate.children.hc_death_rate * state_next.children.hc1_art_pop(0, hd, a, s);
+            intermediate.children.hc_art_grad(0, hd, a, s) -= state_next.children.hc1_art_aids_deaths(0, hd, a, s);
+            state_next.children.hc1_art_pop(0, hd, a, s) += intermediate.children.hc_art_grad(0, hd, a, s);
+          }
+        }else{
+          if(hd < hc_ss.hc2DS){
+            intermediate.children.hc_death_rate = cpars.hc_art_mort_rr(0, a, time_step) * 0.5 * (cpars.hc2_art_mort(hd, 0, a-hc_ss.hc2_agestart) + cpars.hc2_art_mort(hd, 1, a-hc_ss.hc2_agestart));
+            state_next.children.hc2_art_aids_deaths(0, hd, a-hc_ss.hc2_agestart, s)  = intermediate.children.hc_death_rate * state_next.children.hc2_art_pop(0, hd, a-hc_ss.hc2_agestart, s);
+            intermediate.children.hc_art_grad(0, hd, a, s) -= state_next.children.hc2_art_aids_deaths(0, hd, a-hc_ss.hc2_agestart, s);
+            state_next.children.hc2_art_pop(0, hd, a-hc_ss.hc2_agestart, s) += intermediate.children.hc_art_grad(0, hd, a, s);
+          }
+        }
+      }//end ss.NS
+    }// end a
+  }// end ss.hc1DS
+
+  for(int hd = 0; hd < hc_ss.hc1DS; ++hd) {
+    for(int a = 0; a < pars.base.options.p_idx_fertility_first; ++a) {
+      for(int s = 0; s <ss.NS; ++s) {
+        if(a < hc_ss.hc2_agestart){
+          if(state_next.children.hc1_art_pop(1, hd, a, s) > 0){
+            state_next.children.hc1_art_pop(2, hd, a, s) += state_next.children.hc1_art_pop(1, hd, a, s);
+            state_next.children.hc1_art_pop(1, hd, a, s) -= state_next.children.hc1_art_pop(1, hd, a, s);
+          }
+        }else{
+          if(state_next.children.hc2_art_pop(1, hd, a-hc_ss.hc2_agestart, s) > 0){
```

Comment:
I think you might have an issue here, this runs up to `hd < hc_ss.hc1DS` where `hc_ss.hc1DS = 7` so final value of `hd = 6`. But `hc2_art_pop` 2nd dimension is size `hc2DS` which is `6` so this will be trying to index data which at `hd=6` which doesn't exist. Think this might be causing your build failure too.